code/CO2_monitor.ino

```diff
@@ -60,14 +65,24 @@
 
 #include <ESP8266WiFi.h>          // also allows to explicitely turn WiFi off
 #if WIFI_ENABLED
-  #include <Hash.h>               // for SHA1 algorith (for Font Awesome)
-  #include <ESPAsyncTCP.h>
-  #include <ESPAsyncWebServer.h>
-  #include "Webpageindex.h"       // webpage content, same folder as .ino file
+  #if WIFI_WEBSERVER
+    #include <Hash.h>               // for SHA1 algorith (for Font Awesome)
+    #include <ESPAsyncTCP.h>
+    #include <ESPAsyncWebServer.h>
+    #include "Webpageindex.h"       // webpage content, same folder as .ino file
+  #endif
+
+  #if WIFI_MQTT
+    #include <ESP8266HTTPClient.h>
```

Comment:
If I understand for example Ref. [1] correctly, this library is not needed (i.e. this line could potentially be deleted) - but I'm not familiar with mqtt, so I look forward to your weekend research project :) 

[1] https://techtutorialsx.com/2017/04/09/esp8266-connecting-to-mqtt-broker/